src/orion/core/worker/producer.py

```diff
@@ -55,6 +56,23 @@ def produce(self, pool_size, timeout=60, retry_interval=1):
         with self.experiment.acquire_algorithm_lock(
             timeout=timeout, retry_interval=retry_interval
         ) as algorithm:
+            if (
+                self.experiment.knowledge_base
+                and not self.warm_started
+                and is_warmstarteable(algorithm)
+            ):
+                # TODO: Ugly.
+                similar_trials = self.experiment.knowledge_base.get_related_trials(
```

Comment:
Ugly because knowledge_base is an attribute of experiment and experiment is passed to kb's method?

Reply:
Yeah sort-of!

Reply:
Maybe the experiment object could pass it's ExperimentConfig instead? :thinking:

Reply:
Yeah indeed! Maybe that's nicer: 55f7652c3f0fd7535df59a71d9a6ee08539ffd1c

Reply:
Yeah, it's narrower at least.

Reply:
Can we mark this as resolved?